connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -361,7 +362,7 @@ private ConnectorTaskId parseConnectorTaskId(String key) {
         }
     }
 
-    private void readConnectorStatus(String key, byte[] value) {
+    private synchronized void readConnectorStatus(String key, byte[] value) {
```

Comment:
Same question about why this is necessary. It looks like the only part that touches shared state was already protected?